packages/host/tests/integration/search-index-test.ts

```diff
@@ -3710,5 +3749,48 @@ posts/ignore-me.json
         [`${paths.url}books/1`],
       );
     });
+
+    test(`can search for cards by using the 'contains' filter`, async function (assert) {
+      let { data: matching } = await indexer.search({
+        filter: {
+          contains: { title: 'ca' },
+        },
+      });
+      assert.equal(matching.length, 5);
+      assert.deepEqual(
+        matching.map((m) => m.id),
+        [
+          `${paths.url}card-1`,
+          `${paths.url}cards/1`,
+          `${paths.url}cards/2`,
+          `${paths.url}person-card1`,
+          `${paths.url}person-card2`,
+        ],
+      );
+    });
+
+    test(`can search on specific card by using 'contains' filter`, async function (assert) {
+      let { data: matching } = await indexer.search({
+        filter: {
+          on: { module: `${testModuleRealm}person`, name: 'Person' },
+          contains: { title: 'ca' },
+        },
+      });
+      assert.equal(matching.length, 2);
+      assert.deepEqual(
+        matching.map((m) => m.id),
+        [`${paths.url}person-card1`, `${paths.url}person-card2`],
+      );
+    });
+
+    test(`cannot retrieve any cards if 'contains' filter value is null`, async function (assert) {
```

Comment:
we should be able to retrieve cards that have a field value of `null` though, right? maybe include that in the test

Reply:
Updated https://github.com/cardstack/boxel/pull/443/commits/33cf2fc0cbb17ae830817ecd29cf0ef1fbb8f528